ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs

```diff
@@ -434,6 +448,187 @@ traceLedgerProcessing
       mapM_ Debug.traceMarkerIO traces
       return $ newLedger
 
+{-------------------------------------------------------------------------------
+  Analysis: maintain a ledger state and time the five major ledger calculations
+  for each block:
+
+  0. Forecast.
+  1. Header tick.
+  2. Header application.
+  3. Block tick.
+  4. Block application.
+
+  We focus on these 5 operations because they are involved in:
+
+  - Chain syncing.
+  - Block forging.
+  - Block validation.
+
+-------------------------------------------------------------------------------}
+
+data BenchmarkLedgerOpsState blk = BenchmarkLedgerOpsState {
+       -- | RTS stats that correspond to either the point in time in which the
+       -- benchmarking process started, or the end of the previous block
+       -- application.
+       prevRtsStats    :: !GC.RTSStats
+       -- | Intermediate ledger state to which the next block will be applied.
+     , prevLedgerState :: !(ExtLedgerState blk)
+     }
+
+benchmarkLedgerOpsOutputPath :: String
+benchmarkLedgerOpsOutputPath = "ledger-ops-costs.csv" -- Do not forget to update the documentation if you update this.
+
+benchmarkLedgerOps ::
+  forall blk.
+     ( HasAnalysis blk
+     , LedgerSupportsProtocol blk
+     )
+  => Analysis blk
+benchmarkLedgerOps AnalysisEnv {db, registry, initLedger, cfg, limit} =
+    -- TODO: at the moment we do not offer a configuration option for changing
+    -- the location of the output file.
+    IO.withFile benchmarkLedgerOpsOutputPath IO.WriteMode $ \outFileHandle -> do
```

Comment:
Why not just `stdout`? I think that's much simpler and easier to anticipate.

Reply:
Because then one has to fiddle with re-directions and has interference from other processes that print to stdout, like `cabal`. I find it more ergonomic this way, provided we document correctly where the CSV file is stored. What do you think?

Reply:
Hmm. Maybe include a mandatory `--outfile` argument and give `--outfile -` the common `stdout` semantics?

Reply:
Done.